### DISCUSSION TITLE
Encrypted sub claim

### DISCUSSION THREAD

**Discussion by Max-Eriksson** [ActiveLogin/ActiveLogin.Authentication#362] [Q&A] [answered]
Hi,

I've been digging around in the docs to see if you can configure activelogin to encrypt the sub claim, is that possible?

Why would we like to this?

We're using activelogin together with Azure AD B2C and in the AD we're creating [alternative security keys](https://docs.microsoft.com/en-us/azure/active-directory-b2c/social-transformations#createalternativesecurityid) using the sub claim. Due to GDPR we're investigating the possibility to prevent the AD from storing "personnummer" (sub) and an idea would be if activelogin could encrypt the sub with a key only known to identityserver and activelogin. 

BR,
Max

**Comment by PeterOrneholm** [comment, marked as answer]
The direct answer to your question related to Active Login:
There is no (and won't be) any built-in support for encrypting the sub claim. What is available is the option to alter the claim issuing pipeline. We have full docs on that here:
https://docs.activelogin.net/articles/bankid.html#claims-issuing

An input to what you want to achieve:
I would probably implement the solution in a different way, not relying on encrypting the PIN (that has flaws). But architecting a secure auth flow with Azure AD B2C is out of scope for this OSS project - but we would be more than happy to consult in this area. Ping us if you need guidance.
